### PR TITLE
Add screenshot to testresult for XML report

### PR DESCRIPTION
**Pull request by igorrecioh**
[[L143-L145]](https://github.com/igorrecioh/FlaUI/blob/3dafb2092f333a6928eae6d2a9fbd8bad341440b/src/FlaUI.TestUtilities/FlaUITestBase.cs#L143-L145) I have added the AddTestAttachment method in order to include taken screenshot to the report and, therefore, the screenshot will be available in XML report, generated at the end of execution. This is interesting when parsing XML to generate, for example, an HTML report.

[[L278-L286]](https://github.com/igorrecioh/FlaUI/blob/3dafb2092f333a6928eae6d2a9fbd8bad341440b/src/FlaUI.TestUtilities/FlaUITestBase.cs#L278-L286) I have extracted from TakeScreenShot the generation of the full path so that I can use it in previously explained code.

### CODE REVIEW

**Comment by igorrecioh** [comment]
Could anyone provide some input on this? If it is not desire, please comment and I will close PR

**Comment by igorrecioh** [comment]
Hi @tomersalem @Roemer , could you please review this? If it is not a desired feature, let me know and I will close it. Thanks a lot

**Review by tomersalem — APPROVED** [review]

**Comment by igorrecioh** [comment]
Thanks a lot, @tomersalem ! I have just updated the branch with master. I may need another review for this again, sorry.

**Review by tomersalem — CHANGES_REQUESTED** [review]
I like to keep previous results run, please consider adding:
`
/// <summary>
/// static member which holds the current execution date and time
/// </summary>
        private static string _testDateTime = DateTime.Now.ToString("yyyyMMddHHmmss");`

and modify:
`
protected virtual string TestsMediaPath => $@"c:\temp\testsmedia\{TestContext.CurrentContext.Test.Name}\{_testDateTime}";
`

**Review by tomerbyair — APPROVED** [review]

**Review by tomersalem — APPROVED** [review]

**Review by tomersalem — APPROVED** [review]
Thanks,
Tomer

**Comment by igorrecioh** [comment]
@tomersalem It is failing some UI Test, need to check why. I hope this afternoon could be solved

**Comment by igorrecioh** [comment]
@tomersalem It was failing due to ```{TestContext.CurrentContext.Test.Name}````as it contained some illegal characters. Now it's being sanitize to avoid that

**Review by tomersalem — APPROVED** [review]
👍

File: src/FlaUI.TestUtilities/FlaUITestBase.cs [hunk 1]
```diff
@@ -55,11 +55,16 @@ public abstract class FlaUITestBase
...
+        /// static member which holds the current execution date and time
+        /// </summary> 
+        private static string _testDateTime = DateTime.Now.ToString("yyyyMMddHHmmss");
+
         /// <summary>
         /// Path of the directory for the screenshots and videos for the tests.
         /// Defaults to c:\temp\testsmedia.
         /// </summary>
-        protected virtual string TestsMediaPath => @"c:\temp\testsmedia";
+        protected virtual string TestsMediaPath => $@"c:\temp\testsmedia\{SanitizeFileName(TestContext.CurrentContext.Test.Name)}\{_testDateTime}";
```
Comment (Roemer) [inline src/FlaUI.TestUtilities/FlaUITestBase.cs:67]: This is kind of a breaking change but I guess most people anyway overwrite this path so we should be fine.

**Review by Roemer — APPROVED** [review]

**Comment by Roemer** [comment]
Thank you for your contribution. The change has been merged.

**Pull request merged**